gcc/rust/expand/rust-macro-builtins-asm.cc

```diff
@@ -307,9 +307,9 @@ parse_reg_operand_in (InlineAsmContext inline_asm_ctx)
 
       // TODO: When we've succesfully parse an expr, remember to clone_expr()
       // instead of nullptr
-      struct AST::InlineAsmOperand::In in (reg, nullptr);
-      reg_operand.set_in (in);
-      inline_asm_ctx.inline_asm.operands.push_back (reg_operand);
+      // struct AST::InlineAsmOperand::In in (reg, nullptr);
+      // reg_operand.set_in (in);
+      // inline_asm_ctx.inline_asm.operands.push_back (reg_operand);
```

Comment:
so we no longer create any of those structs? is that intentional?

Reply:
in the original code, I created those structs there regardless if i managed to parse expr successfully or not (hence me passing in the nullptr). With the new safeguard, these line will trigger rust_asserts() on nullptr (which is what we want since we don't want to pass around nullptr)

The construction of these structs doesn't matter for now since there isn't any information being passed into the structs (we parse expr via parse_format_string() for now)

Reply:
that makes sense, thanks for the explanation!